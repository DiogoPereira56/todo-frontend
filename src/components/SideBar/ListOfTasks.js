import { PropTypes } from 'prop-types';
import { Li, P } from './SideBar.styles';

const ListOfTasks = ({
    lists,
    setChangeLayout,
    setRename,
    setShowOptions,
    setShowAllTasks,
    orderByTitle,
    setSearchIsActive,
    order,
    setCurrentTaskPage,
    loadListInfo,
}) => {
    function changeActiveList(list) {
        loadListInfo({
            variables: {
                idList: list.idList,
                idClient: list.idClient,
                limit: 13,
                offset: 0,
                orderByTitle: orderByTitle,
                order: order,
            },
        });
    }

    const makeActiveList = (list) => {
        changeActiveList(list);
        setShowAllTasks(false);
        setChangeLayout(false);
        setRename(false);
        setShowOptions(false);
        setSearchIsActive(false);
        setCurrentTaskPage(1);
    };

    return (
        <div data-testid="tasks">
            {lists.map((list) => (
                <Li key={list.idList} onClick={() => makeActiveList(list)}>
                    <P>{list.listName}</P>
                </Li>
            ))}
        </div>
    );
};

ListOfTasks.propTypes = {
    lists: PropTypes.array.isRequired,
    setChangeLayout: PropTypes.func,
    setRename: PropTypes.func,
    setShowOptions: PropTypes.func,
    setShowAllTasks: PropTypes.func,
    orderByTitle: PropTypes.bool,
    setSearchIsActive: PropTypes.func,
    order: PropTypes.string,
    setCurrentTaskPage: PropTypes.func,
    loadListInfo: PropTypes.func,
};

export default ListOfTasks;
