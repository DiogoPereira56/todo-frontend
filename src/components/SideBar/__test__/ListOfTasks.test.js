/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import ListOfTasks from '../ListOfTasks';
import ReactDOM from 'react-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const mockGetLists = () => {
    return [
        { idClient: 3, idList: 1, listName: 'lista', __typename: 'ListOfTasks' },
        { idClient: 3, idList: 2, listName: 'compras', __typename: 'ListOfTasks' },
    ];
};

describe('<ListOfTasks />', () => {
    it('renders without crashing', () => {
        const mockLists = mockGetLists();
        const div = document.createElement('div');
        ReactDOM.render(<ListOfTasks lists={mockLists}></ListOfTasks>, div);
    });

    it('matches ListOfTasks snapshot', () => {
        const mockLists = mockGetLists();
        const tree = renderer.create(<ListOfTasks lists={mockLists}></ListOfTasks>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
