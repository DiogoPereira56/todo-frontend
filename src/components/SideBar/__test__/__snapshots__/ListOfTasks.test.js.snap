// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfTasks /> matches ListOfTasks snapshot 1`] = `
<div
  data-testid="tasks"
>
  <li
    className="sc-eCssSg hfHFtt"
    onClick={[Function]}
  >
    <p
      className="sc-jSgupP bepHJl"
    >
      lista
    </p>
  </li>
  <li
    className="sc-eCssSg hfHFtt"
    onClick={[Function]}
  >
    <p
      className="sc-jSgupP bepHJl"
    >
      compras
    </p>
  </li>
</div>
`;
